inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/convolution_gpu_b_fs_yx_fsv4_int8.cl

```diff
@@ -0,0 +1,163 @@
+KERNEL(convolution_gpu_b_fs_yx_fsv4_int8)(
+    const __global INPUT0_PACKED_TYPE* input,
+    __global OUTPUT_TYPE* output,
+    const __global FILTER_TYPE* weights,
+#if BIAS_TERM
+    const __global BIAS_TYPE* bias,
+#endif
+#if HAS_FUSED_OPS_DECLS
+    FUSED_OPS_DECLS,
+#endif
+    uint split_idx)
+{
+#define AS_TYPE_N_(type, n, x) as_##type##n(x)
+#define AS_TYPE_N(type, n, x) AS_TYPE_N_(type, n, x)
+#define AS_INPUT0_TYPE_4(x) AS_TYPE_N(INPUT0_TYPE, 4, x)
+#define AS_FILTER_TYPE_4(x) AS_TYPE_N(FILTER_TYPE, 4, x)
+
+    const uint items_per_EU = 2;
+    const uint idx = items_per_EU * ((uint)get_global_id(0) * SUB_GROUP_SIZE + (uint)get_global_id(2));
+    const uint idy = (uint)get_global_id(1);
+    uint filter_idx = 0;
+    uint output_idx = 0;
+    uint input_idx = 0;
+    int batch = 0;
+    const uint packed_values = 4;
+#if FILTER_OFM_NUM > 8
+#define FILTER_OFM_MAX 8
+#else
+#define FILTER_OFM_MAX FILTER_OFM_NUM
+#endif
+    __attribute__((opencl_unroll_hint(FILTER_OFM_NUM / FILTER_OFM_MAX)))
+        for (int iter = 0; iter < FILTER_OFM_NUM / FILTER_OFM_MAX + (FILTER_OFM_NUM % FILTER_OFM_MAX != 0); iter++) {
+            int out1[FILTER_OFM_MAX] = { 0 };
+            int out2[FILTER_OFM_MAX] = { 0 };
+            filter_idx = FILTER_OFM_MAX * iter * packed_values;
+
+            __attribute__((opencl_unroll_hint(1)))
+                for (int ifm = 0; ifm < (FILTER_IFM_NUM + 3) / 4; ifm++) {
+                    __attribute__((opencl_unroll_hint(FILTER_SIZE_Y)))
+                        for (int yy = 0; yy < FILTER_SIZE_Y; yy++) {
+                            uint inp_idx = ifm * (INPUT0_FEATURE_PITCH)+(idy + yy) * (INPUT0_Y_PITCH)+idx;
+
+                            __attribute__((opencl_unroll_hint(FILTER_SIZE_X)))
+                                for (int xx = 0; xx < FILTER_SIZE_X; xx++) {
+                                    char8 tmp = as_char8(vload2(0, (__global uint*)(input + inp_idx + xx)));
+
+                                    __attribute__((opencl_unroll_hint(FILTER_OFM_MAX)))
+                                        for (int ofm = 0; ofm < FILTER_OFM_MAX; ofm++) {
+                                            __global uint* www = weights + (filter_idx + ofm * packed_values);
+                                            char4 w = AS_FILTER_TYPE_4(www[0]);
+                                            out1[ofm] = IMAD(out1[ofm], AS_INPUT0_TYPE_4((char4)(tmp[0], tmp[1], tmp[2], tmp[3])), AS_FILTER_TYPE_4(w));
+                                            out2[ofm] = IMAD(out2[ofm], AS_INPUT0_TYPE_4((char4)(tmp[4], tmp[5], tmp[6], tmp[7])), AS_FILTER_TYPE_4(w));
+                                        }
+                                    filter_idx += (packed_values * SUB_GROUP_SIZE);
+                                }
+                        }
+                }
+
+            __attribute__((opencl_unroll_hint(FILTER_OFM_MAX / 4)))
+                for (int ofm = 0; ofm < FILTER_OFM_MAX; ofm += packed_values) {
```

Comment:
Shouldn't leftovers be checked here? Looks like redundant data may be written in this loop
For example for `FILTER_OFM_NUM = 12` at the second `iter` iteration all of 8 values from `out1` and `out2` will be written, but only 4 are needed

Reply:
Right. Done.